apps/o2-blocks/src/github-issue-template/editor.js

```diff
@@ -0,0 +1,154 @@
+	return (
+		<El { ...attrs } className={ classNames( 'wp-block-a8c-github-template', className ) }>
+			{ typeof icon === 'string' ? (
+				<div className="github-template__icon">{ icon }</div>
+			) : (
+				<Icon className="github-template__icon" icon={ icon || githubIcon } />
+			) }
+			<span className="github-template__title">{ title }</span>
+			<div className="github-template__sub-title">{ subTitle }</div>
+			{ body && <span className="github-template__body">{ body }</span> }
+		</El>
+	);
+};
+
+const Edit = ( {
+	onChangeTitle,
+	onChangeUserOrOrg,
+	onChangeRepoName,
+	onChangeBody,
+	title,
+	userOrOrg,
+	repo,
+	body,
+} ) => {
+	return (
+		<Shell
+			className="is-edit"
+			title={ <TextControl onChange={ onChangeTitle } value={ title } /> }
+			subTitle={
+				<>
+					<TextControl onChange={ onChangeUserOrOrg } value={ userOrOrg } />
+					&nbsp;/&nbsp;
+					<TextControl onChange={ onChangeRepoName } value={ repo } />
+				</>
+			}
+			body={ <TextareaControl onChange={ onChangeBody } value={ body } /> }
+		/>
+	);
+};
+
+const View = ( { title, userOrOrg, repo, ...rest } ) => {
+	return (
+		<Shell
+			title={ `${ __( 'Create Issue:' ) } ${ title }` }
+			subTitle={ `${ userOrOrg }/${ repo }` }
+			{ ...rest }
+		/>
+	);
+};
+
+const Invalid = () => (
+	<Shell
+		className="is-warning"
+		title={ __( 'You must provide an org and repo!' ) }
+		subTitle={ __(
+			'Please click here to re-configure to avoid the block from not rendering in the frontend'
+		) }
+		icon="⚠"
+	/>
+);
+
+registerBlockType( 'a8c/github-issue-template', {
+	title: __( 'Github Issue Template', 'a8c' ),
+	icon: githubIcon,
+	category: 'layout',
+	attributes: {
+		userOrOrg: {
+			type: 'string',
+		},
+		repo: {
+			type: 'string',
+		},
+		title: {
+			type: 'string',
+		},
+		body: {
+			type: 'string',
+		},
+	},
+	edit: ( { setAttributes, attributes, attributes: { userOrOrg, repo }, isSelected } ) => {
+		const handlers = {
+			onChangeUserOrOrg( newUserOrOrg ) {
+				setAttributes( { userOrOrg: newUserOrOrg } );
+			},
+
+			onChangeRepoName( newRepo ) {
+				setAttributes( { repo: newRepo } );
+			},
+
+			onChangeTitle( newTitle ) {
+				setAttributes( { title: newTitle } );
+			},
+
+			onChangeBody( newBody ) {
+				setAttributes( { body: newBody } );
+			},
+		};
+
+		const isValid = userOrOrg && repo;
+		const { body, ...viewAttributes } = attributes;
+
+		if ( isSelected ) {
+			return <Edit { ...handlers } { ...attributes } />;
+		}
+
+		return isValid ? <View { ...viewAttributes } /> : <Invalid />;
+	},
+	save: ( { attributes: { userOrOrg, repo, title, body } } ) => {
+		const isValid = userOrOrg && repo;
+		if ( isValid ) {
+			const url = createIssueUrl( {
+				title,
+				repo,
+				user: userOrOrg,
+				body,
+			} );
+
+			const viewAttributes = { title, userOrOrg, repo };
+
+			return <View as="a" attrs={ { href: url } } { ...viewAttributes } />;
```

Comment:
Separating `attrs` here seems strange. Why nest the object instead of passing it along like a normal prop?

```suggestion
			return <View as="a" href={ url } { ...viewAttributes } />;
```

Reply:
I wanted to keep the specific attribute knowledge outside of the Shell component, here: https://github.com/Automattic/wp-calypso/pull/43970/files#diff-442b2be1b7e4a325ade39e41acb7f802R31. However, I can just pass it explicitly here and then keep the same interface for shell by passing it as `attrs` from the View, you're right that it looks better as an explicit prop  👍🏻